### ISSUE TITLE
question about the result with the parameters set as the code 

### ISSUE THREAD

**Issue by fengyisihan** [yaoyao-liu/meta-transfer-learning#5]
I try to run your code without changing any parameters ,but I find the result is far behind reported especially in 5way 1shot. The result is 0.5751+0.0087 in 5 way 1shot and 0.7492+0.0066 in 5way 5shot. Beside, the result of pre-stage is 0.5465. Is this result right in pre-train stage? or any parameters should be changed? Looking forward to your reply.
![image](https://user-images.githubusercontent.com/37177340/64918919-2b8b4a00-d7d7-11e9-90ae-782726d951b6.png)
![捕获](https://user-images.githubusercontent.com/37177340/64918892-ebc46280-d7d6-11e9-9ac9-d5f2ccd53929.PNG)

**Comment by yaoyao-liu** [comment]
We found some little bugs several days ago. Now the bugs are fixed.
Please change `train_aug=True` to `train_aug=False` in Line 21 of `pytorch/dataloader/dataset_loader.py`. Otherwise, the test data will be augmented, which is not correct.

**Comment by fengyisihan** [comment]
Thank you for your quick reply, I check the code but  find   train_aug=False. Maybe not this reason
![image](https://user-images.githubusercontent.com/37177340/64919345-4f9d5a00-d7dc-11e9-857f-41e971ac1766.png)

**Comment by yaoyao-liu** [comment]
<https://github.com/Sha-Lab/FEAT> provides the pre-train model with similar network architecture. You may download the `pth` file in that repo and use it as the pre-train model. If the results are correct with FEAT's pre-train model, you may need to run pre-train phase several times to get a good model. When I am running the code, the pre-train accuracy is about 57.

**Comment by cleardusk** [comment]
Greatly thanks for releasing the code!

Does that mean the final accuracy is determined by the pre-trained model? @yaoyao-liu 
I also pre-trained the model and get about **54.95%** on 5-way 1-shot setting, after meta-train, the 5-way 1/5 shots result are:
```
Val Best Epoch 99, Acc 0.6144, Test Acc 0.5937
Test Acc 0.5937 + 0.0086

Val Best Epoch 96, Acc 0.7747, Test Acc 0.7479
Test Acc 0.7479 + 0.0065
```

My result of 5-way 5-shot is similar to @fengyisihan , while 5-way 1-shot is a little better. But both are worse than `62.1 ± 1.8 /  78.5 ± 0.9` as reported in the `readme.md`.

Another question: the part `Hard task (HT) metabatch` is not released and do you have a plan for it?

**Comment by yaoyao-liu** [comment]
@cleardusk Thanks for your interest in our work. 
1. You may try the pre-train model provided by <https://github.com/Sha-Lab/FEAT>. If it still doesn't work, please let me know. I will upload our pre-train model;
2. For HT meta-batch, we will release the code in the future. We still need time to rewrite the code for publication and run test on it.  I am not sure when will I release it as I am now working for another project. You may also implement HT meta-batch by yourself according to the paper.

**Comment by cleardusk** [comment]
Thanks for your reply, I will try it.

**Comment by yaoyao-liu** [comment]
Weclome. 
I am very sorry that I don't have much time to test the PyTorch version code. If you have any further questions, feel free to add comments in this issue or create another one.

**Comment by cleardusk** [comment]
I try [download_weight.sh](https://github.com/Sha-Lab/FEAT/blob/master/download_weight.sh) in [FEAT](https://github.com/Sha-Lab/FEAT) but fail to download it. 1) My terminal can not use VPN. 2) I copy the url to chrome (used with VPN) but it shows HTTP ERROR 403.
So do you mind to upload your pre-trained model or give a link (Google or Baidu Drive) of the FEAT pre-trained model?

**Comment by fengyisihan** [comment]
I tried the pre-train model in FEAT, but I find worse result Test Acc **0.5760 + 0.0082**
in 1 shot and **0.7381+0.0066** in 5 shot. Could you upload your pre-trained model ? Thanks a lot .
Besides  ,could you help me to check the parameters in meta-training if there is any difference with your training ?
![image](https://user-images.githubusercontent.com/37177340/64937178-d5c4a980-d88b-11e9-9cf1-4896cc62a05b.png)
![image](https://user-images.githubusercontent.com/37177340/64937249-23411680-d88c-11e9-992d-d33e3e27f079.png)

**Comment by yaoyao-liu** [comment]
Here is the pre-train model I use for miniImageNet 1-shot:
[Google Drive](https://drive.google.com/file/d/1uUp9Tn5NVIlZZLWXgvzqVIG8nqLDOHKA/view?usp=sharing)

As the performance of the released pytorch code is not as good as my original one, I'll check the code and hyperparameters in one or two weeks. If you are still not able to get the reported accuracy, please try the tensorflow version. 

Thanks again for your interest. I will add comments to this issue once the bug is fixed. You may also commit the bug you find in the pytorch repo.

**Comment by cleardusk** [comment]
In my case, I use the uploaded pre-trained model but the accuracy is similar to training from scratch in the pre-train phase (very worse). Maybe there is the version conflict or other bugs.
Ignoring the meta-training phase, I am curious how do you train a high-performance model (57% in 5-way 1-shot setting) in the pre-train phase?

**Comment by yaoyao-liu** [comment]
@cleardusk 
For pre-train phase, I run the code many times with different random seeds. 
Besides, I will try to fix the bug when I have time.

**Comment by cleardusk** [comment]
That really depends on the luck or ensembling for the pre-train phase may be a good choice.

**Comment by yaoyao-liu** [comment]
I'll try that. Thanks.

**Comment by cleardusk** [comment]
Just an additional comment, I use an own pre-trained model trained longer and the 1/5 shot results are:
```
Val Best Epoch 43, Acc 0.6157, Test Acc 0.5797
Test Acc 0.5797 + 0.0087

Val Best Epoch 84, Acc 0.7716, Test Acc 0.7621
Test Acc 0.7621 + 0.0064
```

**Comment by yaoyao-liu** [comment]
Thanks for sharing your results. I'll check that.

**Comment by qianyewu** [comment]
> 
> 
> Thanks for your reply, I will try it.

For HT meta-batch, do you have any idea？If you solve it, may you tell me？

**Comment by yaoyao-liu** [comment]
> > Thanks for your reply, I will try it.
> 
> For HT meta-batch, do you have any idea？If you solve it, may you tell me？

@qianyewu 

For HT meta-batch, we will release the code in the future. We still need time to rewrite the code for publication and run test on it. I am not sure when will I release it as I am now working for another project. You may also implement HT meta-batch by yourself according to the paper.

Thanks a lot for your interest in our work. I will let you know once this repo is updated.

**Comment by qianyewu** [comment]
> > > Thanks for your reply, I will try it.
> > 
> > 
> > For HT meta-batch, do you have any idea？If you solve it, may you tell me？
> 
> @qianyewu
> 
> For HT meta-batch, we will release the code in the future. We still need time to rewrite the code for publication and run test on it. I am not sure when will I release it as I am now working for another project. You may also implement HT meta-batch by yourself according to the paper.
> 
> Thanks a lot for your interest in our work. I will let you know once this repo is updated.

 Really thank you！

**Comment by shi1997Yee** [comment]
My result is higher than [cleardusk](https://github.com/yaoyao-liu/meta-transfer-learning/issues/5#issuecomment-531626205)

> 
> ```
> 	pre_train on 5way 1shot : 0.5777
> 	meta_train:
> 	    5way-1shot:
> 		Val Best Epoch 94, Acc 0.6561, Test Acc 0.6183
> 		Test Acc 0.6183 + 0.0079
> 	    5way-5shot:
> 		Val Best Epoch 52, Acc 0.8108, Test Acc 0.7810
> 		Test Acc 0.7810 + 0.0062
> ```
I want to know where is the reported result  `62.1 ± 1.8 / 78.5 ± 0.9` ?

**Comment by yaoyao-liu** [comment]
> My result is higher than [cleardusk](https://github.com/yaoyao-liu/meta-transfer-learning/issues/5#issuecomment-531626205)
> 
> > ```
> > 	pre_train on 5way 1shot : 0.5777
> > 	meta_train:
> > 	    5way-1shot:
> > 		Val Best Epoch 94, Acc 0.6561, Test Acc 0.6183
> > 		Test Acc 0.6183 + 0.0079
> > 	    5way-5shot:
> > 		Val Best Epoch 52, Acc 0.8108, Test Acc 0.7810
> > 		Test Acc 0.7810 + 0.0062
> > ```
> 
> I want to know where is the reported result `62.1 ± 1.8 / 78.5 ± 0.9` ?

Thanks for reporting your results.

The results `62.1 ± 1.8 / 78.5 ± 0.9` were reported in `README.md`. However, as I don't have time to check the bugs in this issue, I removed them temporarily.

**Comment by leon-liangwu** [comment]
@yaoyao-liu 
Thanks for your work.
Any updates for the accuracy of PyTorch?
I did not obtain good performance similar to @cleardusk's neither.

**Comment by yaoyao-liu** [comment]
Hi @leon-liangwu,
Thanks for your interest in our work.

The results for the PyTorch code reported by different people look very different. E.g., `61.83%` by @shi1997Yee and `59.37%' by @cleardusk. I've noticed this problem, and I am working on it. I'll let you know if the PyTorch version is updated. 

Welcome to report your results and other bugs in this issue. If you have further questions, feel free to add more comments.

**Comment by leon-liangwu** [comment]
@yaoyao-liu 
Thanks for your reply and your excellent work.
My result shows as below:
```
# 1 shot
Val Best Epoch 41, Acc 0.6226, Test Acc 0.5780
Test Acc 0.5780 + 0.0083

# 5 shot
Val Best Epoch 85, Acc 0.7768, Test Acc 0.7535
Test Acc 0.7535 + 0.0065
```

**Comment by yaoyao-liu** [comment]
@leon-liangwu 

Thanks for reporting your results. You may try to use our pre-trained model ([Google Drive](https://drive.google.com/file/d/1uUp9Tn5NVIlZZLWXgvzqVIG8nqLDOHKA/view)) and increase the number of validation tasks. In the current version, the number of validation tasks is set to `600`. You may try to increase the number to e.g. `10000`.

We plan to upload the revised PyTorch code after ECCV's deadline. You'll be informed when it is available.

**Comment by leon-liangwu** [comment]
> My result is higher than [cleardusk](https://github.com/yaoyao-liu/meta-transfer-learning/issues/5#issuecomment-531626205)
> 
> > ```
> > 	pre_train on 5way 1shot : 0.5777
> > 	meta_train:
> > 	    5way-1shot:
> > 		Val Best Epoch 94, Acc 0.6561, Test Acc 0.6183
> > 		Test Acc 0.6183 + 0.0079
> > 	    5way-5shot:
> > 		Val Best Epoch 52, Acc 0.8108, Test Acc 0.7810
> > 		Test Acc 0.7810 + 0.0062
> > ```
> 
> I want to know where is the reported result `62.1 ± 1.8 / 78.5 ± 0.9` ?

Hi @shi1997Yee 
Could you please share the procedure how you get this result?
Thanks a lot.
